data-layers/data-layer-api-caller/src/main/java/com/flipkart/aesop/apicallerdatalayer/upsert/ApiCallerUpsertDataLayerFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.flipkart.aesop.apicallerdatalayer.upsert;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.FactoryBean;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Created by aman.gupta on 12/08/15.
+ */
+public class ApiCallerUpsertDataLayerFactory implements FactoryBean<ApiCallerUpsertDataLayer>
+{
+    private URL url;
+    private JSONObject headers;
```

Comment:
Combine both the factories and just extend the base class for the implementation of different methods for the getObjectType() & getOBject()./